packages/babel-preset-es2015/src/index.js

```diff
@@ -1,42 +1,50 @@
+  if (modules === true || moduleTypes.indexOf(modules) === -1) {
+    throw new Error("Preset es2015 'modules' option must be 'false' to indicate no modules\n" +
+      "or a module type which be be one of ['commonjs', 'amd', 'umd', 'systemjs']");
+  }
+  if (typeof spec !== "boolean") throw new Error("Preset es2015 'spec' option must be a boolean.");
 
   return {
     plugins: [
-      [require("babel-plugin-transform-es2015-template-literals"), {loose: loose}],
+      [require("babel-plugin-transform-es2015-template-literals"), { loose, spec }],
       require("babel-plugin-transform-es2015-literals"),
       require("babel-plugin-transform-es2015-function-name"),
-      require("babel-plugin-transform-es2015-arrow-functions"),
+      [require("babel-plugin-transform-es2015-arrow-functions"), { spec }],
       require("babel-plugin-transform-es2015-block-scoped-functions"),
-      [require("babel-plugin-transform-es2015-classes"), {loose: loose}],
+      [require("babel-plugin-transform-es2015-classes"), { loose }],
       require("babel-plugin-transform-es2015-object-super"),
       require("babel-plugin-transform-es2015-shorthand-properties"),
       require("babel-plugin-transform-es2015-duplicate-keys"),
-      [require("babel-plugin-transform-es2015-computed-properties"), {loose: loose}],
-      [require("babel-plugin-transform-es2015-for-of"), {loose: loose}],
+      [require("babel-plugin-transform-es2015-computed-properties"), { loose }],
+      [require("babel-plugin-transform-es2015-for-of"), { loose }],
       require("babel-plugin-transform-es2015-sticky-regex"),
       require("babel-plugin-transform-es2015-unicode-regex"),
       require("babel-plugin-check-es2015-constants"),
-      [require("babel-plugin-transform-es2015-spread"), {loose: loose}],
+      [require("babel-plugin-transform-es2015-spread"), { loose }],
       require("babel-plugin-transform-es2015-parameters"),
-      [require("babel-plugin-transform-es2015-destructuring"), {loose: loose}],
+      [require("babel-plugin-transform-es2015-destructuring"), { loose }],
       require("babel-plugin-transform-es2015-block-scoping"),
       require("babel-plugin-transform-es2015-typeof-symbol"),
-    ].concat(modules ? [
-      [require("babel-plugin-transform-es2015-modules-commonjs"), {loose: loose}],
-    ] : []).concat([
-      [require("babel-plugin-transform-regenerator"), { async: false, asyncGenerators: false }],
-    ]),
+      modules === "commonjs" && [require("babel-plugin-transform-es2015-modules-commonjs"), { loose }],
+      modules === "systemjs" && require("babel-plugin-transform-es2015-modules-systemjs"),
+      modules === "amd" && require("babel-plugin-transform-es2015-modules-amd"),
+      modules === "umd" && require("babel-plugin-transform-es2015-modules-umd"),
+      [require("babel-plugin-transform-regenerator"), { async: false, asyncGenerators: false }]
+    // filter out falsy values
+    ].filter((plugin) => plugin)
```

Comment:
There was that thing about the latest Flow release specifically understanding `.filter(Boolean)`, which I guess means it's no longer particularly obscure?